public/app/plugins/datasource/grafana-azure-monitor-datasource/types/query.ts

```diff
@@ -47,6 +47,7 @@ export interface AzureMetricQuery {
   resourceName?: string;
   /** Resource type */
   metricNamespace?: string;
+  /** used as the value for the metricNamespace param but that name is already used. */
```

Comment:
🤔 That might seem a little odd to someone reading the code and thinking why didn't we refactor then...

What about:
```suggestion
  /** used as the value for the metricNamespace param when different from the resource namespace. */
```

Not sure it's better though... 😅

Reply:
it's probably a good idea to comment on both:

```
  /** metricNamespace is used as the resource type (or resource namespace). It's usually equal to the target metric namespace. Kept the name of the variable as metricNamespace to avoid backward incompatibility issues.  */
  metricNamespace?: string;
  /** customNamespace is used when the resource and metric namespaces are different */
  customNamespace?: string;
```

I would also open an issue to evaluate changing this to fix the naming (with lower priority)